src/main/java/bio/terra/cda/app/controller/QueryApiController.java

```diff
@@ -95,12 +96,36 @@ private String createNextUrl(String jobId, int offset, int limit) {
   }
 
   @Override
-  public ResponseEntity<QueryCreatedData> uniqueValues(String version, String body) {
-
+  public ResponseEntity<QueryCreatedData> uniqueValues(String version, String body, String system) {
     String table = applicationConfiguration.getBqTable() + "." + version;
     NestedColumn nt = NestedColumn.generate(body);
-    String querySql = "SELECT DISTINCT " + nt.getColumn() + " FROM " + table + nt.getUnnestClause();
-    logger.debug("uniqueValues: " + querySql);
+    Set<String> unnestClauses = nt.getUnnestClauses();
+    StringBuffer unnestConcat = new StringBuffer();
```

Comment:
IMO it's clearer to define variables as close to their use as possible. It make their scope smaller and it's easier for a reader to remember what type something is when they're reading the code that uses it. In this case you could move this line down before line 115.

Reply:
Done.